xm-commons-domainevent-db/src/main/java/com/icthh/xm/commons/domainevent/db/service/DatabaseSourceInterceptor.java

```diff
@@ -0,0 +1,261 @@
+        log.info("For domain class {} found corresponded table name {}", entityClass.getSimpleName(), tableName);
+
+        return tableName;
+    }
+
+    private String findTableNameByHibernateNamingStrategy(Object entity) {
+        Metamodel metamodel = entityManager.getMetamodel();
+
+        String tableName = "";
+        if(metamodel instanceof MetamodelImplementor metamodelImplementor) {
+            EntityPersister entityPersister = metamodelImplementor.locateEntityPersister(entity.getClass());
+            if (entityPersister instanceof AbstractEntityPersister abstractEntityPersister) {
+                tableName = abstractEntityPersister.getTableName();
+            }
+        }
+        log.info("Table name by hibernate name strategy: {}", tableName);
+        return tableName;
+    }
+
+    private boolean isIntercepted(String tableName, SourceConfig sourceConfig, JpaEntityContext context) {
+        if (isNull(sourceConfig.getFilter())) {
+            return false;
+        }
+
+        String key = sourceConfig.getFilter().getKey();
+        Boolean needFiltering = databaseFilter.lepFiltering(key, tableName, context);
+
+        if (needFiltering == null) {
+            Map<String, List<EntityFilter>> dsl = sourceConfig.getFilter().getDsl();
+            log.trace("No lep executed, going to process DSL: {}", dsl);
+
+            if (dsl.containsKey(tableName)) {
+                List<EntityFilter> filters = dsl.get(tableName);
+
+                Boolean invoked = lepInvoked(filters, tableName, context);
+
+                return Objects.requireNonNullElseGet(invoked, () -> anyMatchQuery(filters, context));
+            }
+            return false;
+        } else {
+            return needFiltering;
+        }
+    }
+
+    private Boolean lepInvoked(List<EntityFilter> filters, String tableName, JpaEntityContext context) {
+        Boolean invoked = null;
+
+        for (EntityFilter filter : filters) {
+            Boolean result = databaseDslFilter.lepFiltering(filter.getKey(), tableName, context);
+
+            if (TRUE.equals(result)) {
+                return true;
+            } else if (nonNull(result)) {
+                invoked = result;
+            }
+        }
+        return invoked;
+    }
+
+    private boolean anyMatchQuery(List<EntityFilter> filters, JpaEntityContext context) {
+        return filters.stream()
+            .anyMatch(filter -> processQuery(filter.getQuery(), context));
+    }
+
+    private boolean processQuery(Query query, JpaEntityContext context) {
+        log.trace("Query: {}", query);
+        if (isNull(query)) {
+            return false;
+        }
+
+        List<String> propertyNames = Arrays.asList(context.getPropertyNames());
+        log.trace("Property names: {}", propertyNames);
+
+        for (Map.Entry<String, Column> columnEntry : query.getColumns().entrySet()) {
+
+            String queryColumn = columnEntry.getKey();
+            if (propertyNames.contains(queryColumn)) {
```

Comment:
I see iterations over properties several times and finally we convert them to Map. 
Seems it is possible to convert those arrays to map once and then just get by key.
Hypothetically that should improve the performance